app/models/ability.rb

```diff
@@ -0,0 +1,15 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new
+
+    can %i[index show], User
+    can %i[new create show index], Post
+    can %i[new create show index], Comment
+    can %i[update destroy], Post, { author_id: user.id }
+    can %i[update destroy], Comment, { author_id: user.id }
+    can :manage, Post if user.is? :admin
+    can :manage, Comment if user.is? :admin
+  end
```

Comment:
- [ ] As required for the project  user can delete a `post` if it is theirs or if they have an `admin` role and can delete a `comment` if it is theirs or if they have an `admin` role, kindly ensure that your update your app code lines in order to work as expected. Right now, after testing your application where users have different roles(`user` & `admin`), I noticed that `User` can delete both `Comment` and `Post` whereas its/their role is `user`. To help you I suggest you update the code in ability code lines to have logic similar to this and then update other needed files to ensure your application works as expected:
```ruby
class Ability
  include CanCan::Ability

  def initialize(user)

    user ||= User.new

    if user.role == 'admin' 
      can :manage, :all
    else
      can :manage, Post, author: user 
      can :manage, Comment, author: user
      can :read, :all
    end
  end
end
```
Once you have fixed your changes, I suggest you create different users with different `roles` to test if your application is working as expected.

Reply:
Hi man, please please, explain yourself more clearly because I can't understand your english very well, rewrite below what's wrong and what's the changes you require.